boot/freeldr/uefi/uefivid.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * PROJECT:     Freeldr UEFI Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     UEFI Video output
+ * COPYRIGHT:   Copyright 2022 Justin Miller <justinmiller100@gmail.com>
+ */
+
+#include <uefildr.h>
+
+#include <debug.h>
+DBG_DEFAULT_CHANNEL(WARNING);
+
+#define CHAR_WIDTH  8
+#define CHAR_HEIGHT 16
+#define TOP_BOTTOM_LINES 0
+
+/* GLOBALS ********************************************************************/
+
+extern EFI_SYSTEM_TABLE * GlobalSystemTable;
+extern EFI_HANDLE GlobalImageHandle;
+
+UCHAR MachDefaultTextColor = COLOR_GRAY;
+REACTOS_INTERNAL_BGCONTEXT framebufferData;
+
+/* FUNCTIONS ******************************************************************/
+
+VOID
+UefiInitalizeVideo(_In_ EFI_GRAPHICS_OUTPUT_PROTOCOL* gop)
+{
+    gop->SetMode(gop,1);
+
+    framebufferData.BaseAddress        = (ULONG_PTR)gop->Mode->FrameBufferBase;
+    framebufferData.BufferSize         = gop->Mode->FrameBufferSize;
+    framebufferData.ScreenWidth        = gop->Mode->Info->HorizontalResolution;
+    framebufferData.ScreenHeight       = gop->Mode->Info->VerticalResolution;
+    framebufferData.PixelsPerScanLine  = gop->Mode->Info->PixelsPerScanLine;
+    framebufferData.PixelFormat        = gop->Mode->Info->PixelFormat;
+}
+
+VOID
+UefiPrintFramebufferData()
+{
+    TRACE("Framebuffer BaseAddress       : %X\n", framebufferData.BaseAddress);
+    TRACE("Framebuffer BufferSize        : %X\n", framebufferData.BufferSize);
+    TRACE("Framebuffer ScreenWidth       : %d\n", framebufferData.ScreenWidth);
+    TRACE("Framebuffer ScreenHeight      : %d\n", framebufferData.ScreenHeight);
+    TRACE("Framebuffer PixelsPerScanLine : %d\n", framebufferData.PixelsPerScanLine);
+    TRACE("Framebuffer PixelFormat       : %d\n", framebufferData.PixelFormat);
+}
+ULONG
```

Comment:
```suggestion

static ULONG
```
static if that function is only used within this file.